src/http/http_rpc_context.h

```diff
@@ -261,9 +263,19 @@ namespace http
       response_headers[std::string(name)] = value;
     }
 
-    virtual bool response_is_error() const override
+    virtual void set_apply_writes(bool apply) override
```

Comment:
Do you reckon we could have a flag on the handler as well? So that app developers can enforce this statically too (e.g. "I want my handler always apply the transaction, even on failure").

Reply:
We could do this, but I think duplicating ways to set these things is just adding a complication for now. Arguably several of the options we set per-handler (or in this case, per-request) should actually have adjustable per-frontend defaults. But I think we should build a generic solution for all of them rather than adding multiple ways to set _some_.

Reply:
And in this case, its definitely something that could be done within the app itself (add a decorator like our `json_adapter` that always calls `set_apply_writes(true)`), so maybe not something we need to add on our side.